src/auth/types/user.go

```diff
@@ -0,0 +1,51 @@
+package types
+
+import (
+	"github.com/consensys/quorum-key-manager/pkg/errors"
+	manifest "github.com/consensys/quorum-key-manager/src/manifests/types"
+)
+
+// UserInfo are extracted from request credentials by authentication middleware
+type UserInfo struct {
+	// AuthMode records the mode that succeeded to Authenticate the request ('tls', 'api-key', 'oidc' or '')
+	AuthMode string
+
+	// Tenant belonged by the user
+	Tenant string
+
+	// Username identifies the user
+	Username string
+
+	// Roles indicates the user's membership
+	Roles []string
+
+	// Permissions specify
+	Permissions []Permission
+}
+
+func (ui *UserInfo) CheckAccess(mnf *manifest.Manifest) error {
+	if len(mnf.AllowedTenants) == 0 {
+		return nil
+	}
+
+	if ui.Tenant == "" {
+		return errors.NotFoundError("missing credentials")
```

Comment:
I'm not sure it's NotFound here, I think here it's Unauthorized because credentials are missing (401).